Extensions/TileMap/collision/TransformedTileMap.ts

```diff
@@ -41,14 +41,28 @@ namespace gdjs {
         this._source = source;
         this.tag = tag;
         this._layers = new Map<integer, TransformedCollisionTileMapLayer>();
-        for (const sourceLayer of source.getLayers()) {
+        this._buildLayersFromTileMap(source, this._layers);
+      }
+
+      updateFromTileMap(tileMap: TileMapHelper.EditableTileMap) {
+        this._source = tileMap;
+        this._layers = new Map<integer, TransformedCollisionTileMapLayer>();
+
+        this._buildLayersFromTileMap(tileMap, this._layers);
+      }
+
+      _buildLayersFromTileMap(
+        tileMap: TileMapHelper.EditableTileMap,
+        layers: Map<integer, TransformedCollisionTileMapLayer>
```

Comment:
Why do you need to pass `layers` as a parameter? You use it twice with `this._layers` so you could replace it with that in the function (like it was before)

Reply:
it's to satisfy typescript that is not happy with having a member (`_layers`) typed as always defined but that is not set directly in the constructor